analysis/src/CompletionBackEnd.ml

```diff
@@ -451,6 +451,52 @@ let processLocalModule name loc ~prefix ~exact ~env
         (Printf.sprintf "Completion Module Not Found %s loc:%s\n" name
            (Loc.toString loc))
 
+let processLocalInclude includePath _loc ~prefix ~exact ~(env : QueryEnv.t)
+    ~(localTables : LocalTables.t) =
+  (* process only values for now *)
+  localTables.valueTable
+  |> Hashtbl.iter (fun (name, _) (declared : Types.type_expr Declared.t) ->
+         (* We check all the values if their origin is the same as the include path. *)
+         match declared.modulePath with
+         | ModulePath.ExportedModule {name = exportedName}
```

Comment:
@cristianoc, this part was a bit weird.
When I have only:

```
module M = {
  let lex = (a: int) => "foo"
}

module N = {
  include M

  let a = 4
  // let o = a.l
  //            ^com

  // let _ = l
  //          ^com
}
```

in the test file, it it not needed.

Having the entire `tests/analysis_tests/tests/src/IncludeModuleCompletion.res` made the match `ModulePath.ExportedModule` case necessary.
I can't explain this.

Reply:
It's indeed almost unexplainable.
Turns out that when the value table is populated, each value declared in `env` is processed, and the pair (`name, position`) updated in the table.
Included values exist alongside the original values. So there are two values `lex`: one from `M` with module path `ExportedModule`, and one from `include M` with module path `IncludedModule`.
The last second one processed overwrites the first one.
And the order seems kind of arbitrary and changes with small changes to the source file.

Interestingly, when including values from other files, since we use the pair (`name, position`), there could be accidental clash in the `position`, which is just a pair of numbers.

Reply:
@nojaf if included values go to a separate table in `LocalTables.ml`, is it going to work? Or is there a need for them to be in the value table?

Reply:
A separate table could work, so a new one based on `ModulePath.IncludedModule _`?